laws/shared/src/main/scala/algebra/laws/IsSerializable.scala

```diff
@@ -0,0 +1,17 @@
+package algebra.laws
+
+sealed trait IsSerializable[+T]
```

Comment:
Should this be covariant? I don't think so. For instance, if `Int` has a given `IsSerialializable` instance, I can't in general use that same thing for `Any`.

On the other hand, if I want to test a given type `M1` which I know is a subclass of `M2`, then if I have a `Gen[M1]` that I use for all the tests for `M2` (note Gen is covariant), then in that case the `IsSerializable` instance will work to "upcast". I don't immediately see how to formalize that in the type system. Maybe something like:

```scala
  def upcast[M2, M1 <: M2](g: Gen[M1], isser: IsSerializable[M1]): (Gen[M2], IsSerializable[M2]) =
  (g, isser.asInstanceOf[IsSerializable[M2]])
```
But that is only safe if you use the IsSerializable instance with the Gen. It doesn't really glue them together.